### PR TITLE
Fix nightly tests

### LINKED ISSUE #2272: [ISSUE] Upgrade to 0.8.0 version of go-sdk causes regression in `databricks_metastore` delete
The `TestUcAccCreateRecipientDb2DbAws` integration does not pass on master. Git bisect showed the offending commit was the Go SDK migration. I believe TestUcAccMetastoreDataAccessOnAws and TestUcAccMetastoreAssignment are also failing for the same reason.

### Configuration
```hcl
resource "databricks_metastore" "recipient_metastore" {
    name = "<METASTORENAME>-terraform-recipient-metastore"
    storage_root = "s3://deco-uc-prod-aws-us-west-2/test<METASTORENAME>"
    delta_sharing_scope = "INTERNAL"
    delta_sharing_recipient_token_lifetime_in_seconds = "60000"
    force_destroy = true
    lifecycle {
    // fake storage root is causing issues
    ignore_changes = [storage_root] 
    }
}
resource "databricks_recipient" "db2db" {
    name = "<METASTORENAME>-terraform-db2db-recipient"
    comment = "made by terraform"
    authentication_type = "DATABRICKS"
    data_recipient_global_metastore_id = databricks_metastore.recipient_metastore.global_metastore_id
}
```

### Expected Behavior
The test should pass, with the given output:
```
[DEBUG] DELETE /api/2.1/unity-catalog/metastores/<METASTOREID>
> {
>   "force": true
> }
< HTTP/2.0 200 OK
< {}
```

### Actual Behavior
The test fails with the given output:
```
[DEBUG] DELETE /api/2.1/unity-catalog/metastores/<METASTOREID>?force=true
< HTTP/2.0 400 Bad Request
< {
<   "details": [
<     {
<       "@type": "type.googleapis.com/google.rpc.RequestInfo",
<       "request_id": "c9591658-ac38-445e-b243-d9d28aee3090",
<       "serving_data": ""
<     }
<   ],
<   "error_code": "METASTORE_NOT_EMPTY",
<   "message": "Metastore 'c138590b-4836-4a6a-95aa-3958472e86df' is not empty. The metastore has 1 catalog(s), 0... (53 more bytes)"
< }
```

### PR DESCRIPTION
**Pull request by mgyucht**
## Changes
This PR updates the provider to use the patch release of the Go SDK. This corrects the unexpected behavior change where metastore deletion fails after the upgrade.

Once the Go SDK is published with a fix, we can use the updated dependency and make this into a real PR.

Fixes #2272.

## Tests
- [x] `make test` run locally
- [ ] ~~relevant change in `docs/` folder~~
- [x] covered with integration tests in `internal/acceptance`
- [x] relevant acceptance tests are passing
- [x] using Go SDK

### CODE REVIEW

File: catalog/resource_metastore_test.go [hunk 1]
```diff
@@ -134,7 +134,10 @@ func TestDeleteMetastore(t *testing.T) {
 		Fixtures: []qa.HTTPFixture{
 			{
 				Method:   "DELETE",
-				Resource: "/api/2.1/unity-catalog/metastores/abc?force=false",
+				Resource: "/api/2.1/unity-catalog/metastores/abc",
+				ExpectedRequest: map[string]bool{
+					"force": false,
```
Comment (nfx) [inline catalog/resource_metastore_test.go:139]: `force=false` should not have an effect, logically...

there needs to be a PR to go SDK as well.

File: internal/acceptance/metastore_data_access_test.go [hunk 2]
```diff
@@ -8,7 +8,7 @@ func TestUcAccMetastoreDataAccessOnAws(t *testing.T) {
 	unityWorkspaceLevel(t, step{
 		Template: `
 		resource "databricks_metastore" "this" {
-			name          = "primary"
+			name          = "{var.RANDOM}-primary"
```
Comment (nfx) [inline internal/acceptance/metastore_data_access_test.go:11]: let's add random as suffix - will be easier to visually distinguish what is test junk and what is for dev purposes

**Review by nfx — CHANGES_REQUESTED** [review]

File: catalog/resource_metastore_test.go (hunk 1 above)
Comment (mgyucht) [reply catalog/resource_metastore_test.go:139]: SDK PR is here: https://github.com/databricks/databricks-sdk-go/pull/383. This is a draft until that is merged.

File: internal/acceptance/metastore_data_access_test.go (hunk 2 above)
Comment (mgyucht) [reply internal/acceptance/metastore_data_access_test.go:11]: I'll actually move this to a separate PR

**Review by nfx — APPROVED** [review]

**Pull request merged**